domain-http/interface/src/main/java/org/jboss/as/domain/http/server/cors/CorsUtil.java

```diff
@@ -53,10 +53,10 @@ public static boolean isCoreRequest(HeaderMap headers) {
      */
     public static String matchOrigin(HttpServerExchange exchange, Collection<String> allowedOrigins) throws Exception {
         HeaderMap headers = exchange.getRequestHeaders();
-        String origin = headers.getFirst(Headers.ORIGIN);
+        String origin = sanitizeDefaultPort(headers.getFirst(Headers.ORIGIN));
```

Comment:
You're right.
I have also added the possibility to define '*' as allowed-origin.